crescent/commands/options.py

```diff
@@ -0,0 +1,225 @@
+    GroupDMChannel: ChannelType.GROUP_DM,
+    GuildCategory: ChannelType.GUILD_CATEGORY,
+    GuildNewsChannel: ChannelType.GUILD_NEWS,
+    GuildStoreChannel: ChannelType.GUILD_STORE,
+    GuildStageChannel: ChannelType.GUILD_STAGE,
+}
+
+
+def _get_channel_t(t: Type[PartialChannel]) -> list[ChannelType]:
+    valid: list[ChannelType] = []
+    for k, v in _CHANNEL_TYPE_MAP.items():
+        if issubclass(k, t):
+            valid.append(v)
+
+    return valid
+
+
+@dataclass
+class ClassCommandOption:
+    type: OptionType
+    description: str
+    default: UndefinedNoneOr[Any]
+    choices: Optional[Sequence[CommandChoice]]
+    channel_types: Optional[Sequence[ChannelType]]
+    min_value: Optional[Union[int, float]]
+    max_value: Optional[Union[int, float]]
+
+    def _gen_option(self, name: str) -> CommandOption:
+        return CommandOption(
+            type=self.type,
+            name=name,
+            description=self.description,
+            is_required=self.default is UNDEFINED,
+            choices=self.choices,
+            channel_types=self.channel_types,
+            min_value=self.min_value,
+            max_value=self.max_value,
+        )
+
+
+D = TypeVar("D")
+IF = TypeVar("IF", int, float)
+ROMB = TypeVar("ROMB", User, Role, Mentionable, bool)
+
+
+@overload
+def option(
+    type: Union[Type[PartialChannel], Sequence[PartialChannel]],
+    description: str = ...,
+) -> InteractionChannel:
+    ...
+
+
+@overload
+def option(
+    type: Union[Type[PartialChannel], Sequence[PartialChannel]],
+    description: str = ...,
+    *,
+    default: D,
+) -> Union[InteractionChannel, D]:
+    ...
+
+
+@overload
+def option(
+    type: Type[ROMB],
+    description: str = ...,
+) -> ROMB:
+    ...
+
+
+@overload
+def option(
+    type: Type[ROMB],
+    description: str = ...,
+    *,
+    default: D,
+) -> Union[ROMB, D]:
+    ...
+
+
+@overload
+def option(
+    type: Type[IF],
+    description: str = ...,
+    *,
+    choices: Optional[Sequence[Tuple[str, IF]]] = ...,
+    min_value: Optional[IF] = ...,
+    max_value: Optional[IF] = ...,
+) -> IF:
+    ...
+
+
+@overload
+def option(
+    type: Type[IF],
+    description: str = ...,
+    *,
+    default: D,
+    choices: Optional[Sequence[Tuple[str, IF]]] = ...,
+    min_value: Optional[IF] = ...,
+    max_value: Optional[IF] = ...,
+) -> Union[IF, D]:
+    ...
+
+
+@overload
+def option(
+    type: Type[str],
+    description: str = ...,
+    *,
+    choices: Optional[Sequence[Tuple[str, str]]] = ...,
+    min_value: Optional[str] = ...,
+    max_value: Optional[str] = ...,
+) -> str:
+    ...
+
+
+@overload
+def option(
+    type: Type[str],
+    description: str = ...,
+    *,
+    default: D,
+    choices: Optional[Sequence[Tuple[str, str]]] = ...,
+    min_value: Optional[str] = ...,
+    max_value: Optional[str] = ...,
+) -> Union[str, D]:
+    ...
+
+
+def option(  # type: ignore
+    ctype: Union[Type[_VALID_TYPES], Sequence[Type[_VALID_CHANNEL_TYPES]]],
```

Comment:
Can you change this variable name? I think its weird to use `ctype` when this has nothing to do with ctypes.